File: CS/DevExpress.Razor/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Collections;
using System.IO;
using DevExpress.Web.ASPxClasses.Internal;
using DevExpress.Razor.Models;
using System.Web.Script.Serialization;

namespace DevExpress.Razor.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View();
        }
        public ActionResult CallbackPanelPartial() {
            ViewData["Data"] = GetSerializationObject();
            return PartialView("CallbackPanelPartial");
        }


        MyData GetSerializationObject() {
            string jsonText = Request.Params["SerializationData"];
            if (string.IsNullOrEmpty(jsonText))
                return null;

            MyData data = null;
            try {
                data = new JavaScriptSerializer().Deserialize<MyData>(jsonText);
            } catch {
                ViewData["ErrorMessage"] = "Incorrect data";
            }

            return data;
        }
    }
}

File: VB/DevExpress.Razor/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Razor.Models
Imports System.Web.Script.Serialization

Namespace DevExpress.Razor.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
            Return View()
        End Function

        Public Function CallbackPanelPartial() As ActionResult
            ViewData("Data") = GetSerializationObject()
            Return PartialView("CallbackPanelPartial")
        End Function

        Private Function GetSerializationObject() As MyData
            Dim jsonText As String = Request.Params("SerializationData")
            If String.IsNullOrEmpty(jsonText) Then Return Nothing
            Dim data As MyData = Nothing
            Try
                data = New JavaScriptSerializer().Deserialize(Of MyData)(jsonText)
            Catch
                ViewData("ErrorMessage") = "Incorrect data"
            End Try

            Return data
        End Function
    End Class
End Namespace
